lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2899,13 +2899,14 @@ void SplitEmitter::emitMLIRModule() {
   // separate output files, and the remaining top-level verbatim SV/ifdef
   // business that needs to go into each file.
   for (auto &op : *rootOp.getBody()) {
-    if (isa<RTLModuleOp>(op) || isa<RTLModuleExternOp>(op) ||
-        isa<InterfaceOp>(op)) {
+    if (isa<RTLModuleOp>(op) || isa<InterfaceOp>(op)) {
       moduleOps.push_back({&op, perFileOps.size(), /*filename=*/{}});
     } else if (isa<VerbatimOp>(op) || isa<IfDefProceduralOp>(op)) {
       perFileOps.push_back(&op);
     } else if (isa<RTLGeneratorSchemaOp>(op)) {
       /* Empty */
+    } else if (isa<RTLModuleExternOp>(op)) {
+      // ignore extern modules for split Verilog emission
```

Comment:
This can be combined with the previous `isa<...>`. However, this whole thing looks like it could be better handled with a `TypeSwitch` given the number of overlapping cases and the fact that every test is an `isa`.

Reply:
Good point, that would clean things up a bit.